python/ray/serve/_private/application_state.py

```diff
@@ -1099,19 +1094,14 @@ def is_ready_for_shutdown(self) -> bool:
             app_state.is_deleted() for app_state in self._application_states.values()
         )
 
-    def _save_checkpoint_func(
-        self, *, writeahead_checkpoints: Optional[Dict[str, ApplicationTargetState]]
-    ) -> None:
+    def save_checkpoint(self) -> None:
         """Write a checkpoint of all application states."""
 
         application_state_info = {
             app_name: app_state.get_checkpoint_data()
             for app_name, app_state in self._application_states.items()
         }
 
-        if writeahead_checkpoints is not None:
-            application_state_info.update(writeahead_checkpoints)
```

Comment:
I removed the `writeahead_checkpoints` functionality after discussing with @edoakes - it sounds this is vestigial from when the Controller was much more async, and there's no functional reason to do it now (the checkpoint can happen after setting the target state, as long as no changes are made to the Ray cluster between them).